### ISSUE TITLE
Keyboard Navigation broken in Grading Rubric panel

### ISSUE THREAD

**Issue by elsume** [Submitty/Submitty#13050]
**Describe the bug**
The Grading Rubric section of the Grading Interface is not completely navigable using only a keyboard. This is an issue for accessibility. Some specific issues I could find were:
- When all grading boxes are collapsed, the keyboard focus skips straight to the comment box at the bottom, meaning it is impossible to access the grading boxes if only using the keyboard.
- When a grading box is open, it is not possible to select or toggle the grading marks with the keyboard.
- In edit mode, the Discard button is focusable, but pressing enter does not seem to do anything.
- Most of the other interactive elements within the grading box do not show any visible focus indicator, so keyboard users cannot tell which element is currently selected.

**Expected behavior**
- Users should be able to navigate and expand/collapse the grading boxes using keyboard controls.
- Users should be able to select and toggle grading marks using keyboard controls.
- The Discard button should be able to function if enter is pressed while it's selected.
- Every interactive element should display a visible focus indicator so keyboard users can always track their current position.

**To Reproduce**
1. Log in as any user who can access the grading interface.
2. Open a gradeable to grade someone's submission.
3. Select the rubric button to open the Rubric Grading panel.
4. All grading boxes should be closed, then attempt to navigate the grading panel to open a grading box --> observe that the keyboard focus skips past the grading boxes and goes directly to the comment section at the bottom.
5. Next, open a grading box using a mouse and then attempt to select and toggle the grading marks --> observe that it is not possible.
6. Next toggle on the Edit Rubric option and see that the editing UI is now visible on the grading box.
7. Focus on the Discard button with keyboard navigation and try to press enter --> observe nothing happens.
8. Try to navigate the rest of the buttons of the editing UI and observe that there's inconsistent accessibility.

**Configuration** 
- OS: macOS Tahoe
- Browser: Chrome

**Additional Context** 
Related to Issue [#13048](https://github.com/Submitty/Submitty/issues/13048)

**Comment by Jha-2022** [comment]
I would Like Solve this issue
